inspire_json_merger/postprocess.py

```diff
@@ -0,0 +1,124 @@
+    Args:
+        merged(dict): Merged document
+        conflicts(list): List of all possible conflicts
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+
+    """
+    conflicts, merged = postprocess_conflicts(conflicts, merged)
+    conflicts_as_json = [json.loads(c.to_json()) for c in conflicts]
+    flat_conflicts_as_json = remove_ordering_from_authors_conflicts(
+        list(itertools.chain.from_iterable(conflicts_as_json)))
+    merged = remove_ordering_from_authors_merged(merged)
+
+    return merged, flat_conflicts_as_json
+
+
+def remove_ordering_from_authors_conflicts(conflicts):
+    """Sorts authors and cleans up ordering information from conflicts."""
+    new_conflicts = []
+    for conflict in conflicts:
+        if '/authors' in conflict["path"]:
+            del conflict["value"][ORDER_KEY]
+        new_conflicts.append(conflict)
+    return new_conflicts
+
+
+def remove_ordering_from_authors_merged(merged):
+    """Sorts authors and cleans up ordering information from merged record."""
+    if 'authors' in merged:
+        merged["authors"].sort(key=itemgetter(ORDER_KEY))
+        for author in merged.get("authors"):
+            del author[ORDER_KEY]
+    return merged
+
+
+def postprocess_conflicts(conflicts, merged):
+    """Postprocessing conflicts to display only useful conflicts.
+
+    Before flattening and serializing to JSON patch, MERGE conflict looks like this:
+    ('MANUAL_MERGE', ('authors',), ( pmap(<ROOT>), pmap(<HEAD>), pmap(<UPDATE>) )),
+    This function takes out all MANUAL_MERGE and do as follows:
+        - Ignore ROOT
+        - Update merged result with HEAD from conflict by inserting the author
+            from HEAD at the right position.
+        - Look for ADD_BACK_TO_HEAD conflict with identical content as HEAD
+            change and remove it as it's already added to HEAD
+    Args:
+        conflicts(list): List of all possible conflicts
+        merged(dict): Merged document
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+    """
+    new_conflicts = []
+    possible_dplicates = []
+    for conflict in conflicts:
+        if conflict[0] == "MANUAL_MERGE" and "authors" in conflict[1]:
+            new_conflict, merged, head = _process_author_manual_merge_conflict(conflict, merged)
+            new_conflicts.append(new_conflict)
+            possible_dplicates.append(head)
+        else:
+            new_conflicts.append(conflict)
+    new_conflicts = _remove_duplicates(new_conflicts, possible_dplicates)
+    return new_conflicts, merged
+
+
+def _remove_duplicates(conflicts, possible_duplicates):
+    new_conflicts = []
+    for conflict in conflicts:
+        if conflict[0] == "ADD_BACK_TO_HEAD" and "authors" in conflict[1] and conflict[2] in possible_duplicates:
```

Comment:
* this is quardratic and so potentially very expensive in case a large number of authors have conflicts. Can't you use a `set` containing only the `ORDER_KEY` instead for `possible_duplicates`?
* issue with `authors` in other fields

Reply:
Order key might differ between UPDATE and HEAD, there might be same order key but with different content inside conflict, which might be removed but should be displayed. I would stay with current implementation for now. If there will be an issue with efficiency we can think on better solution.